lib/routes/kuaishou/profile.ts

```diff
@@ -0,0 +1,122 @@
+import { Route, Data } from '@/types';
+import puppeteer from '@/utils/puppeteer';
+import cache from '@/utils/cache';
+export const route: Route = {
+    name: 'Profile',
+    path: '/profile/:principalId',
+    radar: [
+        {
+            source: ['kuaishou.com/profile/:principalId'],
+            target: '/profile/:principalId',
+        },
+    ],
+    parameters: {
+        principalId: '用户 id, 可在主页中找到',
+    },
+    example: '/kuaishou/profile/3xk46q9cdnvgife',
+    maintainers: ['nczitzk'],
+    url: 'kuaishou.com/profile/:principalId',
+    description: `:::tip
+    The profile page of the user, which contains the user's information, videos, and other information.
+    :::`,
+    handler,
+};
+
+async function handler(ctx) {
+    const { principalId } = ctx.req.param();
+    const browser = await puppeteer();
+    const page = await browser.newPage();
+    await page.goto(`https://www.kuaishou.com`);
+    await new Promise((resolve) => {
+        setTimeout(resolve, 3e3);
+    });
+    const maxRetryCount = 3;
+    let retryCount = 0;
+    let resolve;
+    let userInfo;
+    const promise = new Promise((res) => {
+        resolve = res;
+    });
+    await page.setRequestInterception(true);
+    page.on('request', (req) => {
+        const url = req.url();
+        if (req.resourceType() === 'image' || req.resourceType() === 'media' || req.resourceType() === 'font') {
+            req.abort();
+        } else if (url.includes('/live_api/profile/public')) {
+            const _url = new URL(url);
+            _url.searchParams.set('count', '999');
+            req.continue({
+                url: _url.toString(),
+            });
+        } else {
+            req.continue();
+        }
+    });
+    page.on('response', async (res) => {
+        if (res.ok() && res.url().includes('/live_api/profile/public')) {
+            const resData = await res.json();
+            if (resData.data.list.length > 0) {
+                resolve(resData.data);
+            } else {
+                if (retryCount > maxRetryCount) {
+                    resolve({});
+                }
+                setTimeout(() => {
+                    page.reload().then();
+                    retryCount++;
+                }, 3e3);
```

Comment:
Let the RSS client retry instead of RSSHub

Reply:
A single attempt may not succeed in obtaining the data